core/include/engine/Hamiltonian_Heisenberg.hpp

```diff
@@ -216,4 +219,4 @@ class Hamiltonian_Heisenberg : public Hamiltonian
 
 } // namespace Engine
 
-#endif
\ No newline at end of file
+#endif
```

Comment:
I don't think you intended to change the end of file sequence. To me this looks like you might be on Windows and didn't set auto-clrf for your git?
See https://stackoverflow.com/a/11535140/4069571

Reply:
I will look into the unit tests and the APIs. 

I'm not sure how the end-of-file sequence happened, I'm not on Windows.

Reply:
Regarding the API, this I can add easily, similarly to the API for anisotropy. I assume it would be better to create a new function for setting the cubic anisotropy rather than modifying the uniaxial anisotropy API.

Regarding tests, I can try create something, but I'm not sure about how to do it since I haven't found any test for the uniaxial anisotropy. Are there some tests that I'm missing? If not, could you suggest how to do the tests?

Reply:
> I'm not sure about how to do it since I haven't found any test for the uniaxial anisotropy

According to Codecov, some test should be using uniaxial anisotropy: https://app.codecov.io/gh/spirit-code/spirit/blob/develop/core/src/engine/Hamiltonian_Heisenberg.cpp
which is actually strange, because I don't think we're actually testing it...

Unfortunately the unit-tests are in pretty bad shape, but it would be good if we try to make it better :)

> could you suggest how to do the tests?

You'd need to create a `Hamiltonian_Heisenberg` instance, run its functions on some data and test the output against expected values.

Probably easiest to do that by creating a `State` using `State_Setup` (don't forget to set the desired Hamiltonian parameters and geometry by calling the corresponding API functions)
Then you can call `state->active_image->hamiltonian->Gradient(spins, gradient);` and `...->Energy(spins);` on some example `spins` for which you know the expected output (note: you'll need to `#include <data/State.hpp>`).